File: 2-Basic-Mathematical-Knowledge-for-Algorithms/002-Sum_of_3_Integers/javascript/main.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// 1. 入力処理
rl.on('line', (line) => {
    // trim()で前後の空白を削除し、split(' ')でスペース分割、map(Number)で数値に変換
    const [a1, a2, a3] = line.trim().split(' ').map(Number);

    // 2. 計算処理
    const sum = a1 + a2 + a3;

    // 3. 出力処理
    console.log(sum);

    rl.close();
});

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/002-Sum_of_3_Integers/rust/src/main.rs
use std::io;

fn main() {
    // 1. 入力処理
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let nums: Vec<i32> = input
        .split_whitespace()
        .map(|s| s.trim().parse().expect("Please enter a number"))
        .collect();

    // 2. 計算処理
    let sum: i32 = nums[0] + nums[1] + nums[2];
    // let sum: i32 = nums.iter().sum();

    // 3. 出力処理
    println!("{}", sum);
}
